pgmpy/factors/Factor.py

```diff
@@ -780,7 +780,9 @@ def __ne__(self, other):
         return not self.__eq__(other)
 
     def __hash__(self):
-        variables = sorted(self.variables)
+        var_with_hashes = list(map((lambda var: (hash(var), var)), self.variables))
```

Comment:
@ankurankan I don't think we can. If you see how we are finding the hash value by swapping the axes according to the sorted order list of variables. Otherwise we have to change the code, but I cannot think of any more efficient way for this.